Pomodoro-App/Pomodoro-App/Home/PomodoroViewController.swift

```diff
@@ -0,0 +1,269 @@
+//
+//  ViewController.swift
+//  Pomodoro-App
+//
+//  Created by Humberto Rodrigues on 14/03/23.
+//
+
+import UIKit
+import AVFoundation
+import UserNotifications
+
+final class PomodoroViewController: UIViewController {
+    
+    //MARK: - Private Variables and constants
+    private var controller = PomodoroController()
+    private var pomodoroScreen: PomodoroScreen?
+    private var timer: Timer?
+    private var timerRemaining: Int = 0
+    private var player: AVAudioPlayer?
+    private var alert: Alert?
+    private let center = UNUserNotificationCenter.current()
+    
+    //MARK: - Inits
+    override func loadView() {
+        super.loadView()
+        self.pomodoroScreen = PomodoroScreen()
+        self.view = pomodoroScreen
+    }
+    
+    //MARK: - Life cycle
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        emiteNotificationAuthorization()
+        self.controller.loadHistory()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        callDelegates()
+        NotificationCenter.default.addObserver(self, selector: #selector(updateTimerWhenAppReturnsToForeground), name: NSNotification.Name(Notification.refresh.rawValue), object: nil)
+
+    }
+    
+    //MARK: - Methods
+    private func callDelegates() {
+        self.alert = Alert(controller: self)
+        self.pomodoroScreen?.stopWatchDelegate(delegate: self)
+        self.pomodoroScreen?.pomodoroConfigurationDelegate(delegate: self)
+        self.pomodoroScreen?.checkHistoryDelegate(delegate: self)
+        center.delegate = self
+    }
+    
+    private func emiteNotificationAuthorization() {
+        
+        center.requestAuthorization(options: [.sound, .alert]) { success, error in
+            if let error = error {
+                print("Error to get Authorization \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    private func buildNotification() {
+        
+        if self.pomodoroScreen?.btStart.isEnabled == true {
```

Comment:
` var title = "Pomodoro finalizado"
var body = "Hora do intervalo, descanse um pouco"
var timeInterval = self.controller.getTiming()
var requestIdentifier =  "Pomodoro"


        if !self.pomodoroScreen?.btStart.isEnabled  {
            title = "Intervalo finalizado"
            body = "Vamos começar um novo Pomodoro?"
            timeInterval = self.controller.getTimingInterval()
            requestIdentifier = "Interval"
        }
        

        let content = UNMutableNotificationContent()
            content.title = title
            content.body = body
            content.sound = .default

            let trigger = UNTimeIntervalNotificationTrigger(timeInterval: TimeInterval(timeInterval), repeats: false)
            let request = UNNotificationRequest(identifier: requestIdentifier, content: content, trigger: trigger)

            center.add(request) { error in
                if let error = error {
                    print("Error to build a notification \(error.localizedDescription)")
                }
            }`